src/node_crypto.cc

```diff
@@ -1631,6 +1660,76 @@ void SSLWrap<Base>::IsInitFinished(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+int compar(const void* a, const void* b) {
+  return memcmp(a, b, CNNIC_WHITELIST_HASH_LEN);
+}
+
+
+int IsSelfSigned(X509* cert) {
+  return X509_NAME_cmp(X509_get_subject_name(cert),
+                       X509_get_issuer_name(cert)) == 0;
+}
+
+
+X509* LookforRoot(STACK_OF(X509)* sk) {
+  int i;
+  X509* cert;
+  for(i = 0; i < sk_X509_num(sk); i++) {
```

Comment:
Style issues: space after `for` and is there a reason to not use `int i = 0`?  The variable isn't used outside the loop, ditto for `cert`.